syft/generic/pointers/pointer_tensor.py

```diff
@@ -1,15 +1,15 @@
-import syft
-import torch
-from syft.frameworks.torch.tensors.interpreters import abstract
-from syft.frameworks.torch import pointers
-
-from syft.workers import AbstractWorker
-
 from typing import List
 from typing import Union
 
+import syft
+from syft.frameworks.types import FrameworkShapeType
+from syft.frameworks.types import FrameworkTensor
+from syft.generic.tensor import AbstractTensor
+from syft.generic.pointers import ObjectPointer
+from syft.workers import AbstractWorker
 
-class PointerTensor(pointers.ObjectPointer, abstract.AbstractTensor):
+
+class PointerTensor(ObjectPointer, AbstractTensor):
```

Comment:
Surely this class is still PyTorch specific?

Reply:
I don't think so! I've generalized most torch-specific functionality already and removed the `import torch` from this file's header and all tests still pass on CI. The only things left to handle are the `data` and `grad` properties, which should be pretty easy to handle on their own

Reply:
Cool